src/viam/sdk/components/encoder/encoder.cpp

```diff
@@ -0,0 +1,125 @@
+#include <viam/sdk/components/encoder/encoder.hpp>
+
+#include <google/protobuf/descriptor.h>
+
+#include <viam/api/component/encoder/v1/encoder.grpc.pb.h>
+#include <viam/api/component/encoder/v1/encoder.pb.h>
+
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/components/encoder/client.hpp>
+#include <viam/sdk/components/encoder/server.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource.hpp>
+
+namespace viam {
+namespace sdk {
+
+std::shared_ptr<ResourceServerBase> EncoderSubtype::create_resource_server(
+    std::shared_ptr<SubtypeService> svc) {
+    return std::make_shared<EncoderServer>(svc);
+};
+
+std::shared_ptr<ResourceBase> EncoderSubtype::create_rpc_client(
+    std::string name, std::shared_ptr<grpc::Channel> chan) {
+    return std::make_shared<EncoderClient>(std::move(name), std::move(chan));
+};
+
+std::shared_ptr<ResourceSubtype> Encoder::resource_subtype() {
+    const google::protobuf::DescriptorPool* p = google::protobuf::DescriptorPool::generated_pool();
+    const google::protobuf::ServiceDescriptor* sd =
+        p->FindServiceByName(viam::component::encoder::v1::EncoderService::service_full_name());
+    if (sd == nullptr) {
```

Comment:
(nit) can we change here and elsewhere in this pr as necessary (not sure if we need to update cryvex as well) to `(if !sd)`?

Reply:
Good catch. Forgot about this. Will update in other pr as well.